### PR TITLE
[Notifier] [RocketChat] Fix undefined index for message id

### LINKED ISSUE #40713: [Notifier] [RocketChat] Notifier fails to set message ID after success
**Symfony version(s) affected**: 5.2.*

**Description**
The Notifier tries to read the ID after successfully sending a message to RocketChat but the array key doesn't exist.
```
Notice: Undefined index: message
```
It happens in the class Symfony\Component\Notifier\Bridge\RocketChat\RocketChatTransport in symfony/rocket-chat-notifier/RocketChatTransport which can be found here https://github.com/symfony/rocket-chat-notifier/blob/5.x/RocketChatTransport.php#L95 and for context reasons here the code:
```php
protected function doSend(MessageInterface $message): SentMessage
    {
        ...

        $result = $response->toArray(false);
        if (!$result['success']) {
            throw new TransportException(sprintf('Unable to post the RocketChat message: %s.', $result['error']), $response);
        }

        $success = $response->toArray(false);

        $sentMessage = new SentMessage($message, (string) $this);
        $sentMessage->setMessageId($success['message']['_id']);

        return $sentMessage;
    }
```
The success response I'm getting from RocketChat consists of only:
```
array (size=1)
  'success' => boolean true
```
No `message` key and therefor also no `id` key.

The change was included a while ago with the commit https://github.com/symfony/rocket-chat-notifier/commit/edfd2933fc08192adaf3d067f88fd8fddec0ac0d and I'm not sure if I'm the only one having this issue since it wasn't raised since then, at least I couldn't find any issues here or on Stackoverflow regarding this topic. Maybe I'm in a special case here too since we host our RocketChat ourselves, but the array key shouldn't be accessed without checking if it exists.

**Possible Solution**
Either return the message directly without adding a message ID
```php
return new SentMessage($message, (string) $this);
```
or at least check if that array key exists before accessing it
```php
$sentMessage = new SentMessage($message, (string) $this);

if (array_key_exists('message', $success) && array_key_exists('_id', $success['message'])) {
    $sentMessage->setMessageId($success['message']['_id']);
}

return $sentMessage;
```
or if it's ok to send an empty string as message ID instead of null if it doesn't exist
```php
$sentMessage = new SentMessage($message, (string) $this);
$sentMessage->setMessageId($success['message']['_id'] ?? '');

return $sentMessage;
```
I hope I did not forget anything important.

Best regards, Sinan

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40713
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/RocketChat/RocketChatTransport.php [hunk 1]
```diff
@@ -99,7 +99,10 @@ protected function doSend(MessageInterface $message): SentMessage
         $success = $response->toArray(false);
 
         $sentMessage = new SentMessage($message, (string) $this);
-        $sentMessage->setMessageId($success['message']['_id']);
+
+        if (\array_key_exists('message', $success) && \array_key_exists('_id', $success['message'])) {
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Bridge/RocketChat/RocketChatTransport.php]: Could we do:

```suggestion
        if (isset($success['message']['_id'])) {
```

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/RocketChat/RocketChatTransport.php]: Sure

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/RocketChat/RocketChatTransport.php]: Done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @OskarStark.

**Pull request merged**